### ISSUE TITLE
comments (bugs?) on clearing text in new search for channels and direct messages 

### ISSUE THREAD

**Issue by pedro-nonfree** [RocketChat/Rocket.Chat#6055]
Rocket.Chat Version: 0.51.0
Running Instances: 1
DB Replicaset OpLog: ?
Node Version: (docker deployment)

Tested with electron client for gnu/linux debian

New autocomplete search for channels and privates is great.

1.
Ctrl+P, you get search. When you go to a channel/direct message that is not in cache, after pressing enter deletes the text (perfect). But if you go to another one, and come back to the same, now is cached, and the text is not cleared. This probably happens because the text clearing function is in the part of loading of channel/direct msg (with the animation of loading), and it is not loaded when it is cached (no animation).

2.
I think that when you click the cross to remove the text that is in the search, the text should be cleared. What happens now is that it looses the focus to the search, but the text is not cleared (is still there). Same issue happens when you do Ctrl+P and then ESC.

**Comment by MartinSchoeler** [comment]
@pedro-nonfree  Thanks for reporting this issue. We value your time in helping us to make Rocket.Chat great. We always try to fix confirmed problems as quickly as we can, and it appears that the issue you reported might already be fixed in the latest release. Can you please help us test it against current latest release? Much appreciated. And please remember to backup your data before upgrading, just to be safe. Cheers!

**Comment by pedro-nonfree** [comment]
Hi @MartinSchoeler,

I tested it with demo.rocket.chat (I assume this has the latest release)

1

From search bar when you press enter always clear text (when cached and not cached). OK

2

When there is text in search bar If I press ESC, text is cleared. OK
When there is text in search bar If I click to cross to cancel search should clear text. FAIL, text is still there

3

Extra cases related:
When I introduced text in the search bar:
- If I put focus in another place, decide:
Option 1: the cross must hold near the text indicating that there is a search active
Option 2: the text on search bar is cleared (I recommend this: more simple)
- If I click one of the results because of the search bar, text should clear

4

Extra case non-related to clear text; but to search bar:
- Results of the search are not showing bold when they contain unread messages.
